drupal (new chart)

prydonius opened the pull request:
#### Details

> [Drupal](https://www.drupal.org/) is one of the most versatile open source content management systems on the market.
- Uses the [Drupal Docker image from Bitnami](https://github.com/bitnami/bitnami-docker-drupal)
- Vendors the MariaDB chart from [stable](https://github.com/kubernetes/charts/tree/master/stable/mariadb) (with changes for persistence)
- Uses alpha PV provisioning for persistence
#### Functional testing
1. Deploy the chart: `helm install incubator/drupal`
2. Wait for external IP address to be configured, and pod to be ready
3. Visit http://$SERVICE_IP/ and login with username `user` and password `bitnami`

arschles commented on buildkite-agent-private-git/Chart.yaml:
```diff
@@ -0,0 +1,9 @@
+version: 1.0.0
+description: Buildkite Agent for private Git repositories
+maintainers:
+  - Tim Lucas <tim@buildkite.com>
```
@rimusz aren't you the maintainer of this chart?

arschles commented on buildkite-agent-private-git/manifests/buildkite-agent-rc.yaml:
```diff
@@ -0,0 +1,36 @@
+            - name: ssh-secret
+              mountPath: /mnt/ssh
+            - name: var-run-docker-sock
+              mountPath: /var/run/docker.sock
```
@rimusz can you modify the `README.md` to indicate that this chart assumes all hosts are configured with the Docker socket at `/var/run/docker.sock`?

rimusz replied:
sure will do both

arschles commented on buildkite-agent/Chart.yaml:
```diff
@@ -0,0 +1,8 @@
+version: 1.0.0
+description: Buildkite Agent
+maintainers:
+  - Tim Lucas <tim@buildkite.com>
```
@rimusz same comment here RE you being the maintainer

arschles commented on buildkite-agent/manifests/buildkite-agent-rc.yaml:
```diff
@@ -0,0 +1,29 @@
+              value: "{{ .token }}"
+          volumeMounts:
+            - name: var-run-docker-sock
+              mountPath: /var/run/docker.sock
```
@rimusz same comment here RE adding to the `README.md` to indicate the location of the Docker socket on hosts

rimusz replied:
readme.md for both charts got as per @arschles comment, forgot to mention it
```
Node hosts need to be configured to run with the Docker socket at /var/run/docker.sock
 for this chart to work properly.
```

rimusz replied:
@mboersma can you take a  look so we can finally release it

rimusz commented:
@arschles @mboersma can I get LGTM for this one, as all requested changes are done

Commit:
drupal (new chart)

prydonius commented:
/cc @michelleN @viglesiasce

viglesiasce commented:
Will review this afternoon.

viglesiasce reviewed (changes requested):
The wizard takes a really long time. Can you add a note that they should be very patient. Mine took around 3 minutes.

chrislovecnm commented on incubator/drupal/charts/mariadb/README.md:
````diff
@@ -0,0 +1,122 @@
+
+To install the chart with the release name `my-release`:
+
+```bash
````
Nit pick, but their is console u can use instead of bash highlighting ;)

chrislovecnm commented on incubator/drupal/charts/mariadb/README.md:
```diff
@@ -0,0 +1,122 @@
+
+> *"An emptyDir volume is first created when a Pod is assigned to a Node, and exists as long as that Pod is running on that node. When a Pod is removed from a node for any reason, the data in the emptyDir is deleted forever."*
+
+For persistence of the data you should replace the `emptyDir` volume with a persistent [storage volume](http://kubernetes.io/docs/user-guide/volumes/), else the data will be lost if the Pod is shutdown.
```
Do we even want to have this with emptyDir? I would never recommend deploying Drupal with emptyDir. Do you have a use case for?

prydonius replied:
Sorry for the confusion on this, we just grabbed the MariaDB from [stable](https://github.com/kubernetes/charts/tree/master/stable/mariadb), changed it to use PVCs by default, but didn't update the README. Kubernetes 1.4 should be released soon, and when it is we will update the standalone MariaDB, and I was just planning on updating all the dependent MariaDBs at the same time.

If you prefer though, I can update the dependent MariaDBs READMEs to reflect the PVC changes we made now.

chrislovecnm replied:
Oh craps... Let's get that into an issue.  Would you be able to run with fixing that outside this PR?

prydonius replied:
Yep, I can take care of that outside this :), I assigned #67 to myself.

chrislovecnm commented on incubator/drupal/charts/mariadb/templates/deployment.yaml:
```diff
@@ -0,0 +1,65 @@
+        heritage: "{{ .Release.Service }}"
+    spec:
+      containers:
+      - name: {{ template "fullname" . }}
```
Can you recommend and put in some resource limits for MySQL?

prydonius replied:
I think this is something that should be added to the standalone MariaDB chart, and then we can update the dependent ones. I missed #60 before, I'll look into it!

chrislovecnm commented on incubator/drupal/templates/NOTES.txt:
```diff
@@ -0,0 +1,26 @@
+1. Get the Drupal URL by running:
+
```
Nice!

chrislovecnm commented on incubator/drupal/templates/deployment.yaml:
```diff
@@ -0,0 +1,79 @@
+    spec:
+      containers:
+      - name: {{ template "fullname" . }}
+        image: "{{ .Values.image }}"
```
Please a resource limits as well.

prydonius replied:
Hmm I've been investigating and it looks like the startup uses quite a bit of CPU, due to the wizard (peaked to ~650m), but after it drops to <50m after Drupal is configured.

Given this, what do you think is an appropriate CPU request/limit?

For memory, 512Mi is a sensible request.

chrislovecnm commented on incubator/drupal/templates/deployment.yaml:
```diff
@@ -0,0 +1,79 @@
+      - name: {{ template "fullname" . }}
+        image: "{{ .Values.image }}"
+        imagePullPolicy: {{ default "" .Values.imagePullPolicy | quote }}
+        env:
```
Does the container image allow for tuning Apache? Worker settings, etc?

prydonius replied:
Not currently, but that's a great idea! I'll create an issue in the [bitnami/apache](https://github.com/bitnami/bitnami-docker-apache) to request those options.

chrislovecnm replied:
@prydonius ping me if you want to review what I am talking about.  I have tuning apache before, and I would rather do Nginx with PHP-fpm running beside it.  Apache is popular, but eh.

prydonius replied:
Personally I prefer the separated tier approach too, but we've run into issues with configuring virtual hosts properly when the web server isn't in the same container as the application. Additionally, many of the applications we're configuring make use of `.htaccess` files, another reason it's easier to have the web server and the app in the same container, and also why nginx is a little harder to configure. That's why we've gone with the Apache + mod_php approach in the Bitnami application containers for now.

chrislovecnm reviewed (commented):
Appreciate the PR! Lots is good stuff coming in!

prydonius commented:
Thanks for the reviews @viglesiasce and @chrislovecnm, some really great suggestions!

I'll add a message in NOTES.txt about the long wait time, and change the README to use the console highlighting. I investigated the resource requests, but will need a little help selecting an appropriate CPU request (see comment).

chrislovecnm commented:
Does it start ok if you do say 300m? It all varies under load. Just always good to put a disclaimer in the conf saying that people will need to modify the values based on ther use case.

Commit:
Review changes

Deployment:
- Add CPU/Memory resources options and defaults

README
- `bash` -> `console`
- Document resources

NOTES.txt
- Add notice about long install time

chrislovecnm commented:
@viglesiasce LGTM

Commit:
Set default CPU request to 300m

prydonius commented:
@chrislovecnm added a default resource request of 300m. Since it's a request and not a limit, it should be good. Thanks!

viglesiasce commented:
Looks great. Thanks @prydonius!!!!

rimusz commented:
Helm Classic deprecated, closing this
